parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -53,6 +53,31 @@ configureSwapFile() {
 }
 {{- end}}
 
+{{- if ShouldConfigureHTTPProxy}}
+configureEtcEnvironment() {
+    {{- if HasHTTPProxy }}
+    echo 'HTTP_PROXY="{{GetHTTPProxy}}"' >> /etc/environment
+    echo 'http_proxy="{{GetHTTPProxy}}"' >> /etc/environment
+    {{- end}}
+    {{- if HasHTTPSProxy }}
+    echo 'HTTPS_PROXY="{{GetHTTPSProxy}}"' >> /etc/environment
+    echo 'https_proxy="{{GetHTTPSProxy}}"' >> /etc/environment
+    {{- end}}
+    {{- if HasNoProxy }}
+    echo 'NO_PROXY="{{GetNoProxy}}"' >> /etc/environment
+    echo 'no_proxy="{{GetNoProxy}}"' >> /etc/environment
```

Comment:
nit: this is not idempotent, can we remove those lines with "HTTP_PROXY=" before appending ?

Reply:
I'm going to leave it for now since I don't have enough time. This func should only be called once though